kivy/uix/image.py

```diff
@@ -250,42 +290,70 @@ def texture_update(self, *largs):
         if not self.source:
             self._clear_core_image()
             return
+
+        self._loops = 0
```

Comment:
This line should be moved to after line 303 because we reset `_loops` if `source` exist and it can be found, otherwise `_clear_core_image` will do the job.